shared/teams/team/container.tsx

```diff
@@ -1,156 +1,105 @@
+import * as Types from '../../constants/types/teams'
+import Team, {Sections} from '.'
+import makeRows from './rows'
 
-type OwnProps = Container.RouteProps<{teamname: string}> & {
-  selectedTab: string
-  setSelectedTab: (arg0: string) => void
+type TabsStateOwnProps = Container.RouteProps<{teamID: Types.TeamID}>
+type OwnProps = TabsStateOwnProps & {
+  selectedTab: Types.TabKey
+  setSelectedTab: (tab: Types.TabKey) => void
 }
 
 // keep track during session
 const lastSelectedTabs = {}
 
 const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
-  const teamname = Container.getRouteProps(ownProps, 'teamname', '')
-  if (!teamname) {
+  const teamID = Container.getRouteProps(ownProps, 'teamID', '')
+  if (!teamID) {
     throw new Error('There was a problem loading the team page, please report this error.')
   }
 
   const selectedTab = ownProps.selectedTab || 'members'
 
-  // TODO this shouldn't be implemented as smashing these unrelated states together makes the typing really messy and this hard to understand
   return {
-    ...(selectedTab === 'members' ? memberMapStateHelper(state, {teamname}) : {}),
-    ...(selectedTab === 'invites' ? invitesMapStateHelper(state, {teamname}) : {}),
-    ...(selectedTab === 'subteams' ? subteamsMapStateHelper(state, {teamname}) : {}),
+    _teamnameTodoRemove: Constants.getTeamDetails(state, teamID).teamname,
     selectedTab,
-    teamname,
+    teamDetails: Constants.getTeamDetails(state, teamID),
+    teamID,
+    yourOperations: Constants.getCanPerformByID(state, teamID),
+    yourUsername: state.config.username,
   }
 }
 
-const mapDispatchToProps = (dispatch: Container.TypedDispatch, {setSelectedTab}: OwnProps) => ({
+const mapDispatchToProps = (dispatch: Container.TypedDispatch) => ({
   _loadTeam: (teamname: string) => dispatch(TeamsGen.createGetDetails({teamname})),
-  _setSelectedTab: (teamname: string, selectedTab: string) => {
-    lastSelectedTabs[teamname] = selectedTab
-    setSelectedTab(selectedTab)
-  },
   onBack: () => dispatch(RouteTreeGen.createNavigateUp()),
 })
 
-class Reloadable extends React.PureComponent<Props & {_load: () => void}> {
-  componentDidUpdate(prevProps) {
-    if (this.props.teamname !== prevProps.teamname) {
-      this.props._load()
-      this.props.setSelectedTab(lastSelectedTabs[this.props.teamname] || 'members')
-    }
-  }
-
-  render() {
-    return (
-      <Kb.Reloadable
-        waitingKeys={Constants.teamGetWaitingKey(this.props.teamname)}
-        onReload={this.props._load}
-        reloadOnMount={true}
-      >
-        <Team
-          rows={this.props.rows}
-          selectedTab={this.props.selectedTab}
-          setSelectedTab={this.props.setSelectedTab}
-          teamname={this.props.teamname}
-        />
-      </Kb.Reloadable>
+const Connected = Container.compose(
+  Container.connect(mapStateToProps, mapDispatchToProps, (stateProps, dispatchProps, ownProps: OwnProps) => {
+    const rows = makeRows(
```

Comment:
No need for custom `mapStateToProps` helpers, we can make the rows with these four fields.